oracle/src/feeds/kraken.rs

```diff
@@ -0,0 +1,84 @@
+use super::{get_http, PriceFeed};
+use crate::{config::CurrencyStore, currency::*, Error};
+use async_trait::async_trait;
+use clap::Parser;
+use reqwest::Url;
+use serde_json::Value;
+
+#[derive(Parser, Debug, Clone)]
+pub struct KrakenCli {
+    /// Fetch the exchange rate from Kraken
+    #[clap(long)]
+    kraken_url: Option<Url>,
+}
+
+pub struct KrakenApi {
+    url: Url,
+}
+
+impl Default for KrakenApi {
+    fn default() -> Self {
+        Self {
+            url: Url::parse("https://api.kraken.com/0").unwrap(),
+        }
+    }
+}
+
+fn extract_response<'a>(value: &'a Value) -> Option<&'a str> {
+    value
+        .get("result")?
+        .as_object()?
+        .iter()
+        .last()? // we are only fetching one anyway
+        .1
+        .get("o")?
```

Comment:
For @philippb90 to check: is "Today's opening price" the best option to use? There is also an option to use the "Volume weighted average price" of the last 24 hours

Reply:
The VWAP option is definitely the better one. How do we go about price sources that do not provide a VWAP price?